### ISSUE TITLE
GWT Support

### ISSUE THREAD

**Issue by ruslansennov** [javaslang/javaslang#1393]
To implement GWT support we should take into account the following considerations:
- It is impossible to implement Match API in GWT for now.
  The main problem is `Class::isAssignableFrom` native java method which can't be implemented in GWT.
  See #1389 
- Some core methods also use reflection or non-GWT abilities (seralization/deserialization, 
  out to IO streams, etc.), so we should introduce `@GwtIncompatible` annotation and switch them off. Also we should mark all Match API as incompatible (for now).
- Early tests don't work with current GWT release. It is necessary to pull latest commits from GWT-repo
  and make own local artifact. 
- Also tests run only in IDE now, but not in console.

**Comment by ruslansennov** [comment]
Blocked by #1290

**Comment by danieldietrich** [comment]
Do the sources need to be released in order to _slangify_ GWT?

**Comment by ruslansennov** [comment]
GWT compiler requires sources of all dependent libraries. If someone want to use snapshot version of javaslang in his project, he should do something like this:
```xml
<dependency>
    <groupId>io.javaslang</groupId>
    <artifactId>javaslang</artifactId>
    <version>2.1.0-SNAPSHOT</version>
</dependency>
<dependency>
    <groupId>io.javaslang</groupId>
    <artifactId>javaslang</artifactId>
    <version>2.1.0-SNAPSHOT</version>
    <classifier>sources</classifier>
</dependency>
```

**Comment by danieldietrich** [comment]
Thx!

**Comment by danieldietrich** [comment]
Can this be closed now after we merged #1547 ? Or is there still work to do?

(I know that it is an ongoing task removing usages of GWT incompatible classes...)

**Comment by danieldietrich** [comment]
I think GWT support does not need to be deferred to 3.0.0. If we need to wait for the next stable GWT release, we should move GWT support to a post 2.1.0 release. /cc @ruslansennov

**Comment by ruslansennov** [comment]
> Or is there still work to do?

I see the light at the end of the tunnel. 

* Need to check deployment stage. There is a problem with MANIFEST file for GWT module
* Need move `javaslang-gwt` to default profile and to leave `javaslang-gwt-example` in GWT profile
* Need for more tests
